network/src/commonMain/kotlin/com/wire/kalium/network/api/configuration/ServerConfigApi.kt

```diff
@@ -0,0 +1,21 @@
+package com.wire.kalium.network.api.configuration
+
+import com.wire.kalium.network.utils.NetworkResponse
+import com.wire.kalium.network.utils.wrapKaliumResponse
+import io.ktor.client.HttpClient
+import io.ktor.client.request.get
+
+interface ServerConfigApi {
+    suspend fun fetchServerConfig(serverConfigUrl: String): NetworkResponse<NetworkConfigDTO>
+}
+
+class ServerConfigApiImp(private val httpClient: HttpClient) : ServerConfigApi {
+
+    /**
+     * Fetch remote configuration
+     * @param serverConfigUrl the remote config url
+     */
+    override suspend fun fetchServerConfig(serverConfigUrl: String): NetworkResponse<NetworkConfigDTO> = wrapKaliumResponse {
+        httpClient.get(serverConfigUrl)
+    }
```

Comment:
wondring if it's a good idea to rename `NetworkConfigDTO` to something like `private data class FetchServerConfigResponse`
```suggestion
    override suspend fun fetchServerConfig(serverConfigUrl: String): NetworkResponse<BackendConfig> = wrapKaliumResponse<NetworkConfigDTO> {
        httpClient.get(serverConfigUrl)
    }.mapSuccess { // map the response to BackendConfig }
```
or we can delete `BackendConfig` class and change `AuthenticatedNetworkContainer` to pass `NetworkConfigDTO`
https://github.com/wireapp/kalium/blob/feat/AR-1037_fetch_server_config/network/src/commonMain/kotlin/com/wire/kalium/network/AuthenticatedNetworkContainer.kt#L35
just so we don't have 2 models to the same thing inside `network`

Reply:
Thanks @MohamadJaara , I good suggestion. Did it as you suggested.